ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx

```diff
@@ -139,52 +147,61 @@ export function PreviewSearchPanel() {
     (keywords: string = '', options?: ComponentsContentTypeParams) => {
       setState({ isFetching: true });
       setError(null);
-      search(site, {
-        ...initialSearchParameters,
-        keywords,
-        ...options,
-        filters: { 'content-type': contentTypes?.map((item) => item.id), 'mime-type': mimeTypes }
-      })
-        .pipe(
-          takeUntil(unMount$),
-          switchMap((result) => {
-            const requests: Array<Observable<ContentInstance>> = [];
-            result.items.forEach((item) => {
-              if (item.type === 'Component') {
-                requests.push(fetchContentInstance(site, item.path, contentTypesLookup));
-              }
-            });
-            return requests.length
-              ? forkJoin(requests).pipe(map((contentInstances) => ({ contentInstances, result })))
-              : of({ result, contentInstances: null });
-          })
-        )
-        .subscribe({
-          next: (response) => {
-            setPageNumber(options ? options.offset / options.limit : 0);
-            if (response.contentInstances) {
-              setState({
-                isFetching: false,
-                items: response.result.items,
-                contentInstanceLookup: createLookupTable(response.contentInstances, 'craftercms.path'),
-                count: response.result.total,
-                limit: options?.limit ?? initialSearchParameters.limit
-              });
-            } else {
-              setState({
-                isFetching: false,
-                items: response.result.items,
-                count: response.result.total,
-                limit: options?.limit ?? initialSearchParameters.limit
+      if (allowedTypesData) {
```

Comment:
This puts the state in fetching and never clears it. Overall the approach doesn't make too much sense. Perhaps just search without types like when allowedTypes.length == 0. We should ensure cancellation as there's a chance of overlapping requests.

Reply:
You still have the if statement?

Reply:
that's just to wait for the allowedTypesData to be loaded in the state

Reply:
Waiting is not necessary, you already adjusted to account for null. Going back to the original comment, this puts the state in `fetching` but never clears it. If waiting is desired then this shouldn't be called in the first place.

Reply:
Actually the fetching state is set back to false it both the next and error sections of the subscription
- success: https://github.com/craftercms/studio-ui/pull/3998/files?show-viewed-files=true&file-filters%5B%5D=#diff-0aa4d1c7d6765372b4344f4c540a5ff21fba5f157d2ec625517c46585a00fe65R182
https://github.com/craftercms/studio-ui/pull/3998/files?show-viewed-files=true&file-filters%5B%5D=#diff-0aa4d1c7d6765372b4344f4c540a5ff21fba5f157d2ec625517c46585a00fe65R190

- error:  https://github.com/craftercms/studio-ui/pull/3998/files?show-viewed-files=true&file-filters%5B%5D=#diff-0aa4d1c7d6765372b4344f4c540a5ff21fba5f157d2ec625517c46585a00fe65R198

Reply:
Removed the if condition as it is not needed anymore